### PR TITLE
Testing pack parameters for ReadPixels into client array buffer.

### PR DESCRIPTION
**Pull request by zhenyao**
Currently passing on Chrome.

### CODE REVIEW

File: sdk/tests/conformance2/reading/read-pixels-pack-parameters.html [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    gl.pixelStorei(gl.PACK_SKIP_PIXELS, packParams.skipPixels);
+    gl.pixelStorei(gl.PACK_SKIP_ROWS, packParams.skipRows);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR);
+
+    gl.clearColor(expectedColor[0] / 255.0, expectedColor[1] / 255.0, expectedColor[2] / 255.0, expectedColor[3] / 255.0);
+    gl.clear(gl.COLOR_BUFFER_BIT);
+
+    var actualWidth = packParams.rowLength > 0 ? packParams.rowLength : width;
+
+    var bytesPerPixel = 4; // see readPixels' parameters below, the format is gl.RED, type is gl.UNSIGNED_BYTE
```
Comment (toji) [inline sdk/tests/conformance2/reading/read-pixels-pack-parameters.html]: Shouldn't this be gl.RGBA like above?

**Comment by toji** [comment]
One comment, but otherwise it LGTM. It would be nice to try and take some of the duplicated code in this new test and break it out into common utility functions, but I'm not sure where the best place is to do that. I'll leave it as an optional follow up patch.

Merge when you're ready.

**Comment by zhenyao** [comment]
Thanks.  I'll do some further refactoring to share the code in the next CL.

File: sdk/tests/conformance2/reading/read-pixels-pack-parameters.html [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        // Revisit this if spec goes the other way.
+        wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION);
+        return;
+    }
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR);
+
+    if (!size)
+        return;
+
+    // Check the last pixel of the last row.
```
Comment (kenrussell) [inline sdk/tests/conformance2/reading/read-pixels-pack-parameters.html]: We may want to consider checking more pixels in these pack alignment tests. It'd be useful to check the first and last pixels of multiple rows.

**Comment by kenrussell** [comment]
Looks good after the fact.

File: sdk/tests/conformance2/reading/read-pixels-pack-parameters.html (hunk 2 above)
Comment (zhenyao) [inline sdk/tests/conformance2/reading/read-pixels-pack-parameters.html]: Agree.  I will enhance the tests in followup CLs.

**Pull request merged**